src/main/java/net/geant/nmaas/portal/api/market/ApplicationController.java

```diff
@@ -314,7 +320,15 @@ private void sendMail(Application app, ApplicationStateChangeRequest stateChange
 			UserView owner = modelMapper.map(userService.findByUsername(applicationBase.getOwner()).orElseThrow(() -> new IllegalArgumentException("Owner not found")), UserView.class);
 			mailAttributes.setAddressees(Collections.singletonList(owner));
 		}
-		this.eventPublisher.publishEvent(new NotificationEvent(this, mailAttributes));
+		if (stateChangeRequest.shouldSendNotification()) {
+			List<UserView> users = userService.findAll()
```

Comment:
Let's have the email sent only to active / enabled users. Probably some additional filtering needs to be applied.